### PR TITLE
Punk updates

### PR DESCRIPTION
**Pull request by catherine-2**
Brief comments on the purpose of your changes:

Quick updates to the CryptoPunks tables:

- New table for floor price per day over time in ETH and $
- Updated metadata table so that attributes are in an array vs. string 

**For Dune Engine V2**

I've checked that:

### General checks:
* [ ] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [ ] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [ ] if adding a new model, I added a test
* [ ] the filename is unique and ends with .sql
* [ ] each sql file is a select statement and has only one view, table or function defined  
* [ ] column names are `lowercase_snake_cased`
* [ ] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [ ] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [ ] `coin_id` represents the ID of the coin on coinpaprika.com
* [ ] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [ ] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [ ] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [ ] where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Comment by jeff-dude** [comment]
> @jeff-dude Added one more CryptoPunks table for floor price over time (forked from the listings over time query we merged yesterday) + changed the metadata table format. Let me know of any questions / comments - thank you!

sounds good, will review real quick. to clarify, all of these objects are views and not tables based on how you've configured them. if they are slow on dune, it's likely cause the full query runs each time. is that intended?

**Comment by catherine-2** [comment]
> > @jeff-dude Added one more CryptoPunks table for floor price over time (forked from the listings over time query we merged yesterday) + changed the metadata table format. Let me know of any questions / comments - thank you!
> 
> sounds good, will review real quick. to clarify, all of these objects are views and not tables based on how you've configured them. if they are slow on dune, it's likely cause the full query runs each time. is that intended?

Thank you! Got it. These run pretty quickly so I think either way is fine / can keep as views unless you disagree. Could you share an example file I can reference if we want to convert them into tables?

**Comment by jeff-dude** [comment]
views are good with me, as long as they perform for ya. we haven't had a ton of use cases for non-incremental tables, so if you did convert in the future, you could follow this config in models:
https://github.com/duneanalytics/spellbook/blob/main/models/uniswap/ethereum/uniswap_v3_ethereum_trades.sql#L1

with test applied on ```unique_key``` here:
https://github.com/duneanalytics/spellbook/blob/main/models/uniswap/ethereum/uniswap_ethereum_schema.yml#L153

File: models/cryptopunks/ethereum/cryptopunks_ethereum_floor_price_over_time.sql [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+            , evt_block_number
+            , evt_index
+            , evt_block_time
+            , evt_tx_hash
+    from {{ source('cryptopunks_ethereum','CryptoPunksMarket_evt_PunkOffered') }}
+)
+, all_no_longer_for_sale_events (
+    select  `punkIndex` as punk_id
+            , 'No Longer For Sale' as event_type
+            , case when evt_tx_hash in (select evt_tx_hash from cryptopunks_ethereum.CryptoPunksMarket_evt_PunkBought) then 'Punk Bought'
```
Comment (jeff-dude) [inline models/cryptopunks/ethereum/cryptopunks_ethereum_floor_price_over_time.sql:28]: can you use source() syntax here?

Comment (catherine-2) [reply models/cryptopunks/ethereum/cryptopunks_ethereum_floor_price_over_time.sql]: Ah yes good catch thank you! Just fixed (+ also on the listings over time file)

**Review by jeff-dude — APPROVED** [review]
looks good!

**Comment by jeff-dude** [comment]
also, if i didn't mention on previous PR, the usage of ```unique_key``` likely isn't doing much on these models, since they are views and don't have a unique test applied in schema file. it's not a huge deal on views, just an fyi, but if you wanted to test uniqueness via gh action on future PRs, that would be how

**Comment by catherine-2** [comment]
> 

Ah got it makes sense - thank you @jeff-dude !

**Pull request merged**